File: src/greedy/P1541_잃어버린괄호.java
package greedy;

import java.util.*;

public class P1541_잃어버린괄호 {
    static Scanner sc = new Scanner(System.in);
    static List<Character> expression = new ArrayList();
    static List<Integer> number = new ArrayList();
    static int minusValue = 0;
    static int plusValue = 0;
    public static void main(String[] args){
        String expression = sc.next();
        boolean zeroSkipFlag = true;
        boolean metMinusFlag = false;
        String numberString = "";
        for(int i=0; i<expression.length(); i++){
            char charecter = expression.charAt(i);
            if(charecter == '0' && zeroSkipFlag)
                continue;
            else if(charecter == '+' && !metMinusFlag){
                plusValue += Integer.parseInt(numberString);
                zeroSkipFlag = true;
                numberString = "";
            }
            else if(charecter == '+' && metMinusFlag){
                minusValue += Integer.parseInt(numberString);
                zeroSkipFlag = true;
                numberString = "";
            }
            else if(charecter == '-' && !metMinusFlag){
                plusValue += Integer.parseInt(numberString);
                metMinusFlag = true;
                zeroSkipFlag = true;
                numberString = "";
            }
            else if(charecter == '-' && metMinusFlag){
                minusValue += Integer.parseInt(numberString);
                zeroSkipFlag = true;
                numberString = "";
            }
            else if(i+1 == expression.length() && metMinusFlag){
                minusValue += Integer.parseInt(numberString + charecter);
            }
            else if(i+1 == expression.length() && !metMinusFlag){
                plusValue += Integer.parseInt(numberString + charecter);
            }
            else {
                zeroSkipFlag = false;
                numberString += charecter;
            }
        }
        System.out.println(plusValue - minusValue);
    }
}

File: py/greedy/P1541_잃어버린괄호.py
string = input()

i = 0
minusValue = 0
plusValue = 0
plusString = ""
while i <= len(string):
    if i == len(string):
        plusValue = plusValue + int(plusString)
        break
    if string[i] == '-':
        if plusString != "":
            plusValue = plusValue + int(plusString)
        plusString = ""
        i = i + 1
        sum = 0
        stringNumber = ""
        while i <= len(string):
            if i == len(string):
                minusValue = minusValue + int(stringNumber)
                break
            if string[i] == '-':
                minusValue = minusValue + int(stringNumber)
                i = i-1
                break
            elif string[i] == '+':
                minusValue = minusValue + int(stringNumber)
                stringNumber = ""
            else:
                if stringNumber == "" and string[i] == "0":
                    i = i+1
                    continue
                else:
                    stringNumber = stringNumber + string[i]
            i = i+1

    elif string[i] == '+':
        plusValue = plusValue + int(plusString)
        plusString = ""
    else:
        if plusString == "" and string[i] == "0":
            i = i+1
            continue
        else:
            plusString = plusString + string[i]
    i = i+1


print(plusValue - minusValue)